README.md

```diff
@@ -103,4 +103,139 @@ The `Store` tag can be configured as follows:
         <td>prefix of the keys whose contains the serialized sessions. Those to avoid possible conflicts if the same
   Redis instance is shared between multiple applications. If not specified, the default prefix value is Tomcat.</td>
     </tr>
-</table>
\ No newline at end of file
+</table>
+
+## Release a new version
+
+This project takes advantage of a *github workflow* for the releases, so:
+- each time you perform a **Pull Request** through github the code is compiled and tested using maven, java11 and a containerized version of redis 6;
+- each time you push a new release tag to github, the code is compiled and tested as above and then a new Docker image is generated and multiple tags are pushed to DockerHub.
+
+### Release tags
+
+When you push a new tag in the form of `v<MAJOR>` or `v<MAJOR><MINOR>` or `v<MAJOR>.<MINOR>.<PATCH>` a series of new tags for your Docker image are generated according to the following schema:
+
+| git tag         | generated docker tags |
+|-----------------|-----------------------|
+| v1              | 1, 1.0, 1.0.0, latest |
+| v1-SNAPSHOT     | 1, 1.0, 1.0.0, latest |
+| v1.2            | 1, 1.2, 1.2.0, latest |
+| v1.2-SNAPSHOT   | 1, 1.2, 1.2.0, latest |
+| v1.2.3          | 1, 1.2, 1.2.3, latest |
+| v1.2.3-SNAPSHOT | 1, 1.2, 1.2.3, latest |
+
+In order to help releasing new tags according to this schema, you can add these ***git aliases***:
+
+#### git lasttag 
+
+Lets you find the last tag in the above form that has been released. It fetches the remote tags first.
+
+An example of usage might be this:
+```bash
+$ git lasttag
+v1.2.3
+```
+To add this alias you have to copy-paste this instruction in your terminal:
+```bash
+git config --global alias.lasttag '!f() { git fetch -tp &>/dev/null; git tag -l v* --sort=v:refname | tail -1; }; f'
+```
+
+### git release
+
+This alias requires `git lasttag` alias to be added among yout git aliases.
+It allows you to create a new release tag (it doesn't push it to the associated remote) increasing the 
+patch, minor or major version number according to the flag you provide in input.
+
+Some usage examples might be:
+```bash
+$ git release --usage
+Creates a new tag from the current commit.
+Usage: 
+  git release [--patch] [--snapshot] # creates a new patch release 
+  git release  --minor  [--snapshot] # creates a new minor release  
+  git release  --major  [--snapshot] # creates a new major release 
+  
+$ git release
+Latest tag found: v1.3.5
+New release tag:  v1.3.6  
+
+$ git release --patch
+Latest tag found: v1.3.6
+New release tag:  v1.3.7
+
+$ git release --snapshot
+Latest tag found: v1.3.7
+New release tag:  v1.3.8-SNAPSHOT
+
+$ git release --minor 
+Latest tag found: v1.3.8-SNAPSHOT
+New release tag:  v1.4.0
+
+$ git release --minor --snapshot
+Latest tag found: v1.4.0
+New release tag:  v1.5-SNAPSHOT.0
```

Comment:
removed the automation of the --snapshot flag from the git alias because, as you suggested, it's no sense to push a release tag with the SNAPSHOT keyword in it!